app/javascript/react/components/NewRide.js

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import { Redirect } from 'react-router-dom'
+
+const NewRide = (props) => {
+  const [redirect, setRedirect] = useState(null)
+  const [newRide, setNewRide] = useState({
+    name: "",
+    location: "",
+    image_url: "",
+    description: ""
+  })
+
+  const location = ['','Frontierland', 'Fantasyland', 'Liberty Square', 'Tomorrowland', 'Adventureland', 'Main Street, U.S.A' ]
+
+  const locationOptions = location.map(location => {
+    return (
+      <option key={location} value={location}>
+        {location}
+      </option>
+    )
+  })
+
+  const handleInputChange = (event) => {
+    setNewRide({
+      ...newRide,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+  }
+
+  const handleSubmitNewRide = async (event) => {
+    event.preventDefault()
+    try {
+      const response = await fetch('/api/v1/rides', {
+        method: "POST",
+        credentials: "same-origin",
+        headers: {
+          "Content-Type": "application/json",
+          "Accept": "application/json" 
+        },
+        body: JSON.stringify({ ride: newRide })
+      })
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw(error)
+      }
+      console.log("Ride was added successfully!")
+      const rideBody = await response.json()
+      if (rideBody.ride.id) {
+        setRedirect(rideBody.ride.id)
+      } 
+    } catch(err) {
+      console.error(`Error in fetch: ${err.message}`)
+    }
+    
+  }
+
+  if (redirect !== null) {
+    return <Redirect to={`/rides/${redirect}`} />
+  }
+
+  return (
+    <div className='new-ride-div'>
+      <h1>Add New Ride</h1>
+      <form className='new-ride' onSubmit={handleSubmitNewRide}>
+        <label>
+          Ride Name:
+          <input type="text" name="name" onChange={handleInputChange} value={newRide.name}/>
+        </label>
+
+        <label>
+          Location:
+          <select type="text" name="location" onChange={handleInputChange} value={newRide.location} >
+            {locationOptions}
+          </select>           
+        </label>
+
+        <label>
+          Image_url:
+          <input type="text" name="image_url" onChange={handleInputChange} value={newRide.image_url}/>
```

Comment:
in the future, we shouldn't implement a feature one way only to go back and implement it a better way. that creates too much overhead to keep track of both new features to work on, and a list of features to fix/change. Image uploading should be its own feature since an image is optional and not required to create a record. We can approach building the application incrementally where we first build the form for the basic info, then when ready tackle the image uploading feature in its entirety 

you can leave this here for now as I think some of your group members are working on image uploading